sdk/trace/batch_span_processor.go

```diff
@@ -199,12 +200,15 @@ func (bsp *BatchSpanProcessor) exportSpans() {
 func (bsp *BatchSpanProcessor) processQueue() {
 	defer bsp.timer.Stop()
 
+	ctx := context.Background()
```

Comment:
This raises an eyebrow or two. Are there any triggers induced by other goroutines that might want to cancel this `Context`?

Reply:
There are but they are handled with the stopCh. That being said, I thing you raise an agreed point that this could be cleaned up to use a context as a signal.

When writing this I also thinking by using the context we could introduce an export timeout here.

In the end I didn't want to have the scope of this PR creep into that territory. I think it can be addressed in a separate PR if someone whats to tackle it.

Reply:
AFAIK the only other triggers are handled by the bsp.stopCh. I've added a cancel function to be called when that is selected to be sure if the downstream export is incorrectly asynchronous it will be canceled.

That being said, I realized when working on this that we could probably simplify the workflow a bit if we instead use the context as a termination signal instead of an additional channel. I've not included it here to not overwhelm the scope of the PR, but it is something someone could follow up on in a separate PR for sure.